Faild to build/Install on OpenSuse Tumbleweed (Gem/Source)

I have tried to install Gosu as Gem the native build of Gosu failed. 
After that fail, I try to build the C++ lib from source and got the same error in make.

_Build error message:_
```
compiling ../../src/AudioFileSDLSound.cpp
../../src/AudioFileSDLSound.cpp: In Konstruktor »Gosu::AudioFile::AudioFile(const std::string&)«:
../../src/AudioFileSDLSound.cpp:54:20: Fehler: »runtime_error« ist kein Element von »std«
   54 |         throw std::runtime_error(message);
      |                    ^~~~~~~~~~~~~
../../src/AudioFileSDLSound.cpp:15:1: Anmerkung: »std::runtime_error« ist im Header »<stdexcept>« definiert; haben Sie vergessen, »#include <stdexcept>« zu schreiben?
   14 | #include <mutex>
  +++ |+#include <stdexcept>
   15 | #include <vector>
../../src/AudioFileSDLSound.cpp: In Konstruktor »Gosu::AudioFile::AudioFile(Gosu::Reader)«:
../../src/AudioFileSDLSound.cpp:73:20: Fehler: »runtime_error« ist kein Element von »std«
   73 |         throw std::runtime_error(message);
      |                    ^~~~~~~~~~~~~
../../src/AudioFileSDLSound.cpp:73:20: Anmerkung: »std::runtime_error« ist im Header »<stdexcept>« definiert; haben Sie vergessen, »#include <stdexcept>« zu schreiben?
../../src/AudioFileSDLSound.cpp: In Elementfunktion »ALenum Gosu::AudioFile::format() const«:
../../src/AudioFileSDLSound.cpp:95:16: Fehler: »runtime_error« ist kein Element von »std«
   95 |     throw std::runtime_error("Unsupported number of channels: " + std::to_string(channels) +
      |                ^~~~~~~~~~~~~
../../src/AudioFileSDLSound.cpp:95:16: Anmerkung: »std::runtime_error« ist im Header »<stdexcept>« definiert; haben Sie vergessen, »#include <stdexcept>« zu schreiben?
../../src/AudioFileSDLSound.cpp:97:1: Warnung: Kontrollfluss erreicht Ende von Nicht-void-Funktion [-Wreturn-type]
   97 | }
      | ^
make: *** [Makefile:240: AudioFileSDLSound.o] Fehler 1

make failed, exit code 2

```
After that I have add into src/AudioFileSDLSound.cpp and src/Input.cpp in line 14 `#include <stdexcept>`.
That has fix it for me and I have run a successfully make and make install after the adding of `#include <stdexcept>`.

Code-Blocks form zipped-source(1.4.5) with the adding of include:
_cat ../src/AudioFileSDLSound.cpp_
```
#include <Gosu/Platform.hpp>

#ifndef GOSU_IS_IPHONE

#include "AudioFile.hpp"
#include "AudioImpl.hpp"

#include <SDL_sound.h>

#include <cstdlib>
#include <cstring>
#include <algorithm>
#include <memory>
#include <mutex>
#include <stdexcept>
#include <vector>
....
rest of the code
```
_cat ../src/Input.cpp_
```

#include <Gosu/Platform.hpp>
#if !defined(GOSU_IS_IPHONE)

#include <Gosu/Input.hpp>
#include <Gosu/TextInput.hpp>
#include <Gosu/Utility.hpp>

#include <SDL.h>
#include <utf8proc.h>

#include <cwctype>
#include <cstdlib>
#include <algorithm>
#include <array>
#include <mutex>
#include <stdexcept>
using namespace std;
...
rest of the code file
```

I'm not a c++ programmer and can't say that this a fix but hope that will help that I can build the Gem out of the box on OpenSuSe Tumblweed.

This workaround works for me in the master/main branch and in the zipped source version 1.4.5

**System and libs:**
OS: OpenSuSe Tumbleweed
GCC/++ : 12 / 13
**SDL-Libs:**
```
i+ | libSDL2-2_0-0                   | Paket      | 2.26.4-1.2    | x86_64 | Haupt-Repository (OSS)
i+ | libSDL2-2_0-0                   | Paket      | 2.26.4-1.2    | x86_64 | openSUSE:Tumbleweed
i+ | libSDL2-2_0-0-32bit             | Paket      | 2.26.4-1.2    | x86_64 | Haupt-Repository (OSS)
i+ | libSDL2-2_0-0-32bit             | Paket      | 2.26.4-1.2    | x86_64 | openSUSE:Tumbleweed
i+ | libSDL2_gfx-1_0-0               | Paket      | 1.0.4-2.10    | x86_64 | Haupt-Repository (OSS)
i+ | libSDL2_gfx-1_0-0               | Paket      | 1.0.4-2.10    | x86_64 | openSUSE:Tumbleweed
i+ | libSDL2_gfx-devel               | Paket      | 1.0.4-2.10    | x86_64 | Haupt-Repository (OSS)
i+ | libSDL2_gfx-devel               | Paket      | 1.0.4-2.10    | x86_64 | openSUSE:Tumbleweed
i+ | libSDL2_image-2_0-0             | Paket      | 2.6.3-1.2     | x86_64 | Haupt-Repository (OSS)
i+ | libSDL2_image-2_0-0             | Paket      | 2.6.3-1.2     | x86_64 | openSUSE:Tumbleweed
i+ | libSDL2_mixer-2_0-0             | Paket      | 2.6.3-1.2     | x86_64 | Haupt-Repository (OSS)
i+ | libSDL2_mixer-2_0-0             | Paket      | 2.6.3-1.2     | x86_64 | openSUSE:Tumbleweed
i+ | libSDL2_net-2_0-0               | Paket      | 2.2.0-1.3     | x86_64 | Haupt-Repository (OSS)
i+ | libSDL2_net-2_0-0               | Paket      | 2.2.0-1.3     | x86_64 | openSUSE:Tumbleweed
i+ | libSDL2_ttf-2_0-0               | Paket      | 2.20.2-1.2    | x86_64 | Haupt-Repository (OSS)
i+ | libSDL2_ttf-2_0-0               | Paket      | 2.20.2-1.2    | x86_64 | openSUSE:Tumbleweed
i+ | SDL2-devel                      | Paket      | 2.26.4-1.2    | x86_64 | Haupt-Repository (OSS)
i+ | SDL2-devel                      | Paket      | 2.26.4-1.2    | x86_64 | openSUSE:Tumbleweed
i+ | SDL2_image-devel                | Paket      | 2.6.3-1.2     | x86_64 | Haupt-Repository (OSS)
i+ | SDL2_image-devel                | Paket      | 2.6.3-1.2     | x86_64 | openSUSE:Tumbleweed
i+ | SDL2_mixer-devel                | Paket      | 2.6.3-1.2     | x86_64 | Haupt-Repository (OSS)
i+ | SDL2_mixer-devel                | Paket      | 2.6.3-1.2     | x86_64 | openSUSE:Tumbleweed
i+ | SDL2_net-devel                  | Paket      | 2.2.0-1.3     | x86_64 | Haupt-Repository (OSS)
i+ | SDL2_net-devel                  | Paket      | 2.2.0-1.3     | x86_64 | openSUSE:Tumbleweed
i+ | SDL2_Pango-devel                | Paket      | 2.1.5-1.2     | x86_64 | Haupt-Repository (OSS)
i+ | SDL2_Pango-devel                | Paket      | 2.1.5-1.2     | x86_64 | openSUSE:Tumbleweed
i+ | SDL2_sound-devel                | Paket      | 2.0.1.g17-1.9 | x86_64 | Haupt-Repository (OSS)
i+ | SDL2_sound-devel                | Paket      | 2.0.1.g17-1.9 | x86_64 | openSUSE:Tumbleweed
i+ | SDL2_ttf-devel                  | Paket      | 2.20.2-1.2    | x86_64 | Haupt-Repository (OSS)
i+ | SDL2_ttf-devel                  | Paket      | 2.20.2-1.2    | x86_64 | openSUSE:Tumbleweed
```
**libstd**:
```
i+ | libstdc++-devel                  | Paket | 13-1.2             | x86_64 | Haupt-Repository (OSS)
i+ | libstdc++-devel                  | Paket | 13-1.2             | x86_64 | openSUSE:Tumbleweed
i+ | libstdc++6                       | Paket | 13.0.1+git6995-1.1 | x86_64 | Haupt-Repository (OSS)
i+ | libstdc++6                       | Paket | 13.0.1+git6995-1.1 | x86_64 | openSUSE:Tumbleweed
i+ | libstdc++6-32bit                 | Paket | 13.0.1+git6995-1.1 | x86_64 | Haupt-Repository (OSS)
i+ | libstdc++6-32bit                 | Paket | 13.0.1+git6995-1.1 | x86_64 | openSUSE:Tumbleweed
i+ | libstdc++6-devel-gcc11           | Paket | 11.3.1+git2076-2.3 | x86_64 | Haupt-Repository (OSS)
i+ | libstdc++6-devel-gcc11           | Paket | 11.3.1+git2076-2.3 | x86_64 | openSUSE:Tumbleweed
i+ | libstdc++6-devel-gcc12           | Paket | 12.2.1+git749-2.1  | x86_64 | Haupt-Repository (OSS)
i+ | libstdc++6-devel-gcc12           | Paket | 12.2.1+git749-2.1  | x86_64 | openSUSE:Tumbleweed
i+ | libstdc++6-locale                | Paket | 13.0.1+git6995-1.1 | x86_64 | Haupt-Repository (OSS)
i+ | libstdc++6-locale                | Paket | 13.0.1+git6995-1.1 | x86_64 | openSUSE:Tumbleweed
i+ | libstdc++6-pp                    | Paket | 13.0.1+git6995-1.1 | x86_64 | Haupt-Repository (OSS)
i+ | libstdc++6-pp                    | Paket | 13.0.1+git6995-1.1 | x86_64 | openSUSE:Tumbleweed
i+ | libstdc++6-pp-32bit              | Paket | 13.0.1+git6995-1.1 | x86_64 | Haupt-Repository (OSS)
i+ | libstdc++6-pp-32bit              | Paket | 13.0.1+git6995-1.1 | x86_64 | openSUSE:Tumbleweed

```
**GCC-Version:**
```
i+ | cross-aarch64-gcc13                           | Paket      | 13.0.1+git6995-1.1    | x86_64 | Haupt-Repository (OSS)
i+ | cross-aarch64-gcc13                           | Paket      | 13.0.1+git6995-1.1    | x86_64 | openSUSE:Tumbleweed
i+ | gcc                                           | Paket      | 13-1.2                | x86_64 | Haupt-Repository (OSS)
i+ | gcc                                           | Paket      | 13-1.2                | x86_64 | openSUSE:Tumbleweed
i+ | gcc-c++                                       | Paket      | 13-1.2                | x86_64 | Haupt-Repository (OSS)
i+ | gcc-c++                                       | Paket      | 13-1.2                | x86_64 | openSUSE:Tumbleweed
i+ | gcc11                                         | Paket      | 11.3.1+git2076-2.3    | x86_64 | Haupt-Repository (OSS)
i+ | gcc11                                         | Paket      | 11.3.1+git2076-2.3    | x86_64 | openSUSE:Tumbleweed
i+ | gcc11-c++                                     | Paket      | 11.3.1+git2076-2.3    | x86_64 | Haupt-Repository (OSS)
i+ | gcc11-c++                                     | Paket      | 11.3.1+git2076-2.3    | x86_64 | openSUSE:Tumbleweed
i+ | gcc11-locale                                  | Paket      | 11.3.1+git2076-2.3    | x86_64 | Haupt-Repository (OSS)
i+ | gcc11-locale                                  | Paket      | 11.3.1+git2076-2.3    | x86_64 | openSUSE:Tumbleweed
i+ | gcc12                                         | Paket      | 12.2.1+git749-2.1     | x86_64 | Haupt-Repository (OSS)
i+ | gcc12                                         | Paket      | 12.2.1+git749-2.1     | x86_64 | openSUSE:Tumbleweed
i+ | gcc12-c++                                     | Paket      | 12.2.1+git749-2.1     | x86_64 | Haupt-Repository (OSS)
i+ | gcc12-c++                                     | Paket      | 12.2.1+git749-2.1     | x86_64 | openSUSE:Tumbleweed
i+ | gcc13                                         | Paket      | 13.0.1+git6995-1.1    | x86_64 | Haupt-Repository (OSS)
i+ | gcc13                                         | Paket      | 13.0.1+git6995-1.1    | x86_64 | openSUSE:Tumbleweed
i+ | gcc13-c++                                     | Paket      | 13.0.1+git6995-1.1    | x86_64 | Haupt-Repository (OSS)
i+ | gcc13-c++                                     | Paket      | 13.0.1+git6995-1.1    | x86_64 | openSUSE:Tumbleweed
i+ | gcc13-devel                                   | Paket      | 13.0.1+git6995-1.1    | x86_64 | Haupt-Repository (OSS)
i+ | gcc13-devel                                   | Paket      | 13.0.1+git6995-1.1    | x86_64 | openSUSE:Tumbleweed
i+ | gcc13-locale                                  | Paket      | 13.0.1+git6995-1.1    | x86_64 | Haupt-Repository (OSS)
i+ | gcc13-locale                                  | Paket      | 13.0.1+git6995-1.1    | x86_64 | openSUSE:Tumbleweed
i+ | libgcc_s1                                     | Paket      | 13.0.1+git6995-1.1    | x86_64 | Haupt-Repository (OSS)
i+ | libgcc_s1                                     | Paket      | 13.0.1+git6995-1.1    | x86_64 | openSUSE:Tumbleweed
i+ | libgcc_s1-32bit                               | Paket      | 13.0.1+git6995-1.1    | x86_64 | Haupt-Repository (OSS)
i+ | libgcc_s1-32bit                               | Paket      | 13.0.1+git6995-1.1    | x86_64 | openSUSE:Tumbleweed
i+ | libstdc++6-devel-gcc11                        | Paket      | 11.3.1+git2076-2.3    | x86_64 | Haupt-Repository (OSS)
i+ | libstdc++6-devel-gcc11                        | Paket      | 11.3.1+git2076-2.3    | x86_64 | openSUSE:Tumbleweed
i+ | libstdc++6-devel-gcc12                        | Paket      | 12.2.1+git749-2.1     | x86_64 | Haupt-Repository (OSS)
i+ | libstdc++6-devel-gcc12                        | Paket      | 12.2.1+git749-2.1     | x86_64 | openSUSE:Tumbleweed

```

Sry for the long post.

--- a/src/AudioFileSDLSound.cpp
+++ b/src/AudioFileSDLSound.cpp
@@ -13,6 +13,7 @@
 #include <memory>
 #include <mutex>
 #include <vector>
+#include <stdexcept>
 
 struct Gosu::AudioFile::Impl : private Gosu::Noncopyable
 {

--- a/src/AudioFileSDLSound.cpp
+++ b/src/AudioFileSDLSound.cpp
@@ -12,8 +12,8 @@
 #include <algorithm>
 #include <memory>
 #include <mutex>
-#include <vector>
 #include <stdexcept>
+#include <vector>
 
 struct Gosu::AudioFile::Impl : private Gosu::Noncopyable
 {

